src/MessageList/MessageListItem.vala

```diff
@@ -248,13 +257,26 @@ public class Mail.MessageListItem : Gtk.ListBoxRow {
 
         header_event_box.button_release_event.connect ((event) => {
             expanded = !expanded;
-            return false;
+            return Gdk.EVENT_STOP;
         });
 
         destroy.connect (() => {
             loading_cancellable.cancel ();
         });
 
+        starred_button.button_release_event.connect (() => {
+            if (Camel.MessageFlags.FLAGGED in (int) message_info.flags) {
+                message_info.set_flags (Camel.MessageFlags.FLAGGED, 0);
+                starred_icon.icon_name = "non-starred-symbolic";
+                starred_icon.tooltip_text = _("Star message");
+            } else {
+                message_info.set_flags (Camel.MessageFlags.FLAGGED, ~0);
```

Comment:
What does ~0 means? 🤔

Reply:
It means complement all bits of 0, so it's all bits to 1.
With the `Camel.MessageFlags.FLAGGED` mask it means that only the `Camel.MessageFlags.FLAGGED` flag will be set to 1